### PR TITLE
[next] Fixes #1484: Re-Adds the Search page component

### LINKED ISSUE #1484: [next] Porting Search page component, see src/frontend/src/pages/search.js from Gatsby to Next
**What would you like to be added**:
SearchPage rewritten in TS 
**Why would you like this to be added**:
We want to port this page to Next using TS

### PR DESCRIPTION
**Pull request by chrispinkney**
## Issue This PR Addresses
Fixes #1484 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI
- [x] **Next Port**: Port to NextJS
## Description
This PR adds `search.tsx`, a page which will render the [`<SearchPage />` component](https://github.com/Seneca-CDOT/telescope/pull/1583). It also injects SEO along with the banner and head components.

**EDIT:** The [`<SearchPage />` component](https://github.com/Seneca-CDOT/telescope/pull/1583) was just land-ho'd, so I merged master into this PR, renamed the `search.tsx` component @c3ho to `SearchPage`, added SEO to our search page, the banner, head, and darkmode toggle button.

~~Just waiting on `<SearchPage />` to be finished then I can test this further.~~

~~**CI/CD will fail** as the  `<SearchPage />` component isn't yet implemented, when it gets released I'll (try to remember to) come back and test this further.~~

### How to test
Pull the PR and navigate to [http://localhost:8000/search](http://localhost:8000/search).

firefox_2021-02-04_15-48-23

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/frontend/next/src/pages/search.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import SEO from '../components/SEO';
+import SearchPage from '../components/SearchPage';
+
+const Search = () => {
+  return (
+    <>
+      <SEO pageTitle="Search" />
+      <SearchPage />
+    </>
+  );
```
Comment (tonyvugithub) [inline src/frontend/next/src/pages/search.tsx:10]: Can you add a mock `<SearchPage/>` so we can pass the CI/CD test?

**Review by tonyvugithub — CHANGES_REQUESTED** [review]

File: src/frontend/next/src/pages/search.tsx (hunk 1 above)
Comment (chrispinkney) [reply src/frontend/next/src/pages/search.tsx:10]: I'm going to rebase this once SearchPage lands, I think adding in a mock component will just make that component unnecessarily harder to land

**Review by c3ho — APPROVED** [review]
I'm fine with this, since it just moves some stuff around

**Comment by chrispinkney** [comment]
Looks like Darkmode will need to be fixed for Searchpage. When this lands we'll have to open another issue for that.

**Review by c3ho — APPROVED** [review]

File: src/frontend/next/src/pages/search.tsx [hunk 2]
```diff
@@ -1,63 +1,28 @@
...
-        onSubmit={onSubmitHandler}
-      />
-      <br />
-      <SearchResults text={textParam} filter={filterParam} />
-      <BackToTopButton />
-    </div>
+    <>
+      <Head>
+        <title>Search</title>
+        <meta property="og:title" content="Telescope" key="title" />
```
Comment (humphd) [inline src/frontend/next/src/pages/search.tsx:18]: This isn't dong anything there, let's remove and have it happen in _document.tsx

**Review by humphd — APPROVED** [review]
One minor thing and this is good.  If you want to file an issue to fix that later (it's in other pages too), that's also fine.

**Pull request merged**